### DISCUSSION TITLE
Strange cell behaviour

### DISCUSSION THREAD

**Discussion by mr-manuel** [Louisvdw/dbus-serialbattery#746] [General]
Hello,

anyone of you saw already a similar cell voltage behaviour as I have?

The cell is first reaching max voltage, then the balancer (JKBMS) kicks in and balances the cell (1). After the cells are on the same voltage (2) then the cell is loosing voltage and is then balanced again until all cells reach the same voltage (3) again.

For me it would make more sense, if all voltage rise the same after reaching (2).

![grafik](https://github.com/Louisvdw/dbus-serialbattery/assets/12592765/6df69023-50a7-431c-8e75-0c385edfa10b)

![grafik](https://github.com/Louisvdw/dbus-serialbattery/assets/12592765/0b91afe3-9fb7-48a9-8f99-366d72d4cbfd)

What are your opinions?

**Comment by maxx8888** [comment]
Strange...

Bad Contact? Is the charging current changing a lot at Point2 and bad contact causes wrong Voltage Measurement?
Can you maybe add the charge current in this Graph?

Did you maybe crosscheck with Voltmeter?

**Reply by mr-manuel** [reply]
![grafik](https://github.com/Louisvdw/dbus-serialbattery/assets/12592765/8aabbee9-8ea8-43d3-ba02-bfe00b72a602)

I did not yet cross check with the voltmeter, but I will do it on weekend. Then I will also try to discharge only this cell with a resistor a bit.

**Reply by mr-manuel** [reply]
Now I checked multiple times with different Voltmeters and the voltage is correct. The contacts are also good and under full load there is no abnormal temperature. Everything is under 35° C on 60 A.

**Reply by maxx8888** [reply]
Ok, maybe cell itself? High internal resistance would look like that also, i think. Voltage Drop on the cell would be higher, Balancer would kick in to early and discharge too much. Can you measure internal resistance? Try to load the batteries with less current and check if it is improving

**Comment by majic79** [comment]
What's your balance voltage set at? I don't balance until cell voltages all hit 3.55 - I saw some odd behaviour when I set a lower balance start voltage (3.4) and I think one cell was at a higher state of charge than the others, causing some spikes in charge. Top balancing (balancing only when the voltages hit desired charge level) seems to be the way to solve this for LFP (my spikes have reduced, now I just need to spend longer in the absorption voltage for the balancing to have time to work)

**Reply by mr-manuel** [reply]
I start balancing in the purple area which is `3.40` V. I will try to increase to `3.50` V, since I'm only able to charge to `3.555` V (which is the `64` V charging limit for the MultiPlus-II and `18` cells).

**Reply by majic79** [reply]
I'm finding that matching the max charge voltage in SerialBattery to the Balance Start voltage in JKBMS has (eventually) cleaned up some cell drift that I've been seeing. You could try setting max charge voltage (in SerialBattery) to 3.50 and then start balance at 3.50 also. You'd need to observe it over a couple of charge cycles to see if the cells charge curves come back together
